src/category_theory/limits/shapes/equalizers.lean

```diff
@@ -172,14 +239,82 @@ abbreviation equalizer := limit (parallel_pair f g)
 abbreviation equalizer.ι : equalizer f g ⟶ X :=
 limit.π (parallel_pair f g) zero
 
-@[simp, reassoc] lemma equalizer.condition : equalizer.ι f g ≫ f = equalizer.ι f g ≫ g :=
+lemma equalizer.ι.fork : fork.ι (limits.limit.cone (parallel_pair f g)) = equalizer.ι f g := rfl
+
+lemma equalizer.ι.eq_app_zero : (limit.cone (parallel_pair f g)).π.app zero = equalizer.ι f g := rfl
+
+@[reassoc] lemma equalizer.condition : equalizer.ι f g ≫ f = equalizer.ι f g ≫ g :=
 begin
   erw limit.w (parallel_pair f g) walking_parallel_pair_hom.left,
   erw limit.w (parallel_pair f g) walking_parallel_pair_hom.right
 end
 
 abbreviation equalizer.lift {W : C} (k : W ⟶ X) (h : k ≫ f = k ≫ g) : W ⟶ equalizer f g :=
 limit.lift (parallel_pair f g) (fork.of_ι k h)
+
+/-- A specialization of the uniqueness statement about limits to equalizers -/
+lemma equalizer.lift.uniq {W : C} (k : W ⟶ X) (h : k ≫ f = k ≫ g) (l : W ⟶ equalizer f g)
+  (i : l ≫ (equalizer.ι f g) = k) : l = (equalizer.lift f g k h) :=
+begin
+  refine is_limit.uniq (limit.is_limit (parallel_pair f g)) (fork.of_ι k h) l _,
+  intro j, cases j,
+  { simp only [fork.of_ι_app_zero, limit.cone_π], exact i },
+  { rw [←cone_parallel_pair_left, fork.of_ι_app_one, ←category.assoc, equalizer.ι.eq_app_zero, i] }
+end
+
+/-- An equalizer morphism is a monomorphism -/
+lemma equalizer.ι_mono : mono (equalizer.ι f g) :=
+{ right_cancellation := λ Z h k w, begin
+  have h₀ : (h ≫ (equalizer.ι f g)) ≫ f = (h ≫ (equalizer.ι f g)) ≫ g :=
+    by simp only [category.assoc, equalizer.condition],
+  rw equalizer.lift.uniq f g (h ≫ (equalizer.ι f g)) h₀ h rfl,
+  rw equalizer.lift.uniq f g (h ≫ (equalizer.ι f g)) h₀ k w.symm
+end }
+end
+
+/-- The identity determines a cone on the equalizer diagram of f and f -/
+def cone_parallel_pair_self : cone (parallel_pair f f) :=
+{ X := X,
+  π :=
+  { app := λ j, match j with | zero := 𝟙 X | one := f end }}
+
+@[simp] lemma cone_parallel_pair_self_π_app_zero : (cone_parallel_pair_self f).π.app zero = 𝟙 X :=
+rfl
+
+lemma cone_parallel_pair_self_X : (cone_parallel_pair_self f).X = X := rfl
```

Comment:
simp?

Reply:
Thank you for all the simp suggestions. I have been reading that simp tagging is more subtle than it looks, so it is nice to get input from someone knowledgeable.